api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -229,7 +229,28 @@ public void setLocation(Location location) {
 		
 		return leaves;
 	}
-	
+
+	/**
+	 * Convenience method to recursively get all leaf obs for a given parent obs.
+	 * @param obsParent indicates the parent Obs
+	 * @param includedVoided indicates whether or not to include voided Obs
+	 * @return a Set of all leaves Obs including the parent Obs
+	 */
+	private Set<Obs> getFlattenObsLeaves(Obs obsParent, boolean includedVoided) {
+		Set<Obs> leaves = new LinkedHashSet<>();
+
+		if (includedVoided || (!includedVoided && !obsParent.getVoided())) {
```

Comment:
Just a thought, i think checking !includedVoided is not necessary . If its coming to the right operand that means its already false.

Reply:
Thanks @dilanthas. That's a subtle and elegant improvement.